lib/output.js

```diff
@@ -354,7 +356,18 @@ function OutputStream(options) {
         force_semicolon : force_semicolon,
         to_ascii        : to_ascii,
         print_name      : function(name) { print(make_name(name)) },
-        print_string    : function(str, quote) { print(encode_string(str, quote)) },
+        print_string    : function(str, quote, escape_directive) {
+            var encoded = encode_string(str, quote);
+            if (escape_directive === true && encoded.indexOf("\\") === -1) {
```

Comment:
I like what you've done here moving the logic out of DEFPRINT(AST_String).  My apologies for the "hackery" comment. It was uncalled for.

Rather than `escape_directive === true`, isn't `escape_directive` sufficient?